### PR TITLE
Fix TimeZoneInfo.HasIanaId when using Local Time Zone

### PR DESCRIPTION
**Pull request by tarekgh**
https://github.com/dotnet/runtime/issues/58326

We have exposed the new property `TimeZoneInfo.HasIanaId` in .NET 6.0. This property tells if the Time Zone object is created with IANA Id. This property returns the correct results except when using `TimeZoneInfo.Local`. The reason is when we initialize TimeZoneInfo.Local property we clone the object before returning it, but we missed cloning `HasIanaId` property inside the returned object. The fix is to ensure `HasIanaId` is initialized correctly in `TimeZoneInfo.Local`.

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs [hunk 1]
```diff
@@ -1954,7 +1955,7 @@ private static TimeZoneInfoResult TryGetTimeZoneFromLocalMachine(string id, bool
                 else
                 {
                     value = new TimeZoneInfo(match!._id, match._baseUtcOffset, match._displayName, match._standardDisplayName,
-                                          match._daylightDisplayName, match._adjustmentRules, disableDaylightSavingTime: false);
+                                          match._daylightDisplayName, match._adjustmentRules, disableDaylightSavingTime: false, match.HasIanaId);
```
Comment (eerhardt) [inline src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs:1958]: Are both of these changes getting tested with the test below? Is it possible to test both of them?

**Review by eerhardt — APPROVED** [review]
The changes LGTM.

File: src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs (hunk 1 above)
Comment (tarekgh) [reply src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs:1958]: I added a test for the second case too.

File: src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs [hunk 2]
```diff
@@ -2640,12 +2640,29 @@ public static void UsingAlternativeTimeZoneIdsTest(string windowsId, string iana
...
 
         public static bool SupportIanaNamesConversion => PlatformDetection.IsNotBrowser && PlatformDetection.ICUVersion.Major >= 52;
+        public static bool SupportIanaNamesConversionAndRemoteExecution => SupportIanaNamesConversion && RemoteExecutor.IsSupported;
+
+        // This test is executed using the remote execution because it needs to run before creating the time zone cache to ensure testing with that state.
+        // There are already other tests that test after creating the cache.
+        [ConditionalFact(nameof(SupportIanaNamesConversionAndRemoteExecution))]
+        public static void IsIanaIdWithNotCacheTest()
+        {
+            Assert.Equal(!s_isWindows, TimeZoneInfo.Local.HasIanaId);
```
Comment (eerhardt) [inline src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs:2650]: I don't see us using RemoteExecutor in this test. Am I missing something?

**Review by eerhardt — APPROVED** [review]

File: src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs (hunk 2 above)
Comment (tarekgh) [reply src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs]: No. I added it. I was testing manaully running this method only when I committed the change. After I added the remote execution I forgot to to push it :-)

**Pull request merged**